app/views/rooms/show.html.erb

```diff
@@ -0,0 +1,49 @@
+<div id="faux-background" class="mobile-content"></div>
+<%= render 'nav' %>
+<hr class="nav-offset not-zoomable">
+
+
+
+<div class="mobile-content main">
+  <% p current_user.show_solutions.to_s %>
+  <% if (current_user.show_solutions && @room.state == "do_not_use")  %>
+      <div id="question-options-evaluate">
+        <div class="question-options-message" id="question-evaluation-next">Tento test v režime Učenie<br>nemáte sprístupnený!</div>
+        <a href="<%=
+            url_for :controller => 'weeks', :action => 'show'
+        %>" class="question-button question-button-red" id="question-evaluation-next" >Zmeň režim</a>
+      </div>
+  <%else%>
+      <% @learning_objects.each do |lo| %>
+          <a href="<%=
+              url_for :controller => 'questions', :action => 'show', :id => lo.url_name, :week_number => @week.id, :room_number => @room.id
+          %>" class="card">
+            <div class="question-card-content">
+              <h3 class="card-header"><%= lo.lo_id %></h3>
+              <p class="card-text">
+                <%= lo.question_text %>
+              </p>
+            </div>
+            <div class="card-info">
+              <%
+                 result = @results.find {|r| r["result_id"] == lo.id.to_s}
+                 unless result.nil?
+                   views = result['visited']
+                   done = result['solved']
+                 end
+                 unless views.nil?
```

Comment:
Vo view by malo byt minimum Ruby/Rails kodu - iba nevyhnutne zlo a vypis obsahu premennych.

Niekedy ale samozrejme potrebujes vykonat aj taky kod, ktory controller nepotrebuje, ale potrebuje ho view - presne takyto pripad. Inzinieri, ktori vyvijali Rails na to mysleli, a preto vynasli helpery :)
http://mixandgo.com/blog/the-beginner-s-guide-to-rails-helpers